src/main/java/health/ere/ps/service/idp/BearerTokenService.java

```diff
@@ -39,13 +41,27 @@ public class BearerTokenService {
 
 
     @PostConstruct
-    public void init() {
+    public void init() throws Exception {
+        // Make this process sync to prevent Application start in dysfunctional state
+
+        List<Integer> retrySeconds = appConfig.getIdpInitializationRetriesSeconds();
+        int retryPeriodMs = appConfig.getIdpInitializationPeriodMs();
+        boolean initialized = Retrier.callAndRetry(retrySeconds, retryPeriodMs, this::initializeIdp, bool -> bool);
```

Comment:
This will block the whole start thread for a long time if it does not work. The retrier should use an own thread.

Reply:
```
if it is not possible to reach the Idp at the start up of ere-health an error is logged and the whole application is discfunctional
```
That's why I did it this way because there is no reason to start the app, right?